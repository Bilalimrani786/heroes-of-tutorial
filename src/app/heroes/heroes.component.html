<h2>{{heroTitle}}</h2>
<ul class="heroes">
  <a *ngFor="let hero of heroes" routerLink="/detail/{{hero.id}}">
    <span class="badge">{{hero.id}}</span> {{hero.name}}
  </a>
</ul>