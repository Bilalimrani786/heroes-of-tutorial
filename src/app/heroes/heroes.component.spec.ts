import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterModule, Routes } from '@angular/router';
import { HeroesComponent } from './heroes.component';
import { FormsModule } from '@angular/forms';
import { APP_BASE_HREF } from '@angular/common';

describe('HeroesComponent', () => {
  let component: HeroesComponent;
  let fixture: ComponentFixture<HeroesComponent>;
  const routes: Routes = [
    { path: '', redirectTo: '/dashboard', pathMatch: 'full' },
    { path: 'heroes', component: HeroesComponent }
  ];
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HeroesComponent],
      imports: [FormsModule, RouterModule.forRoot(routes)],
      providers: [{ provide: APP_BASE_HREF, useValue: '/' }]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HeroesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    const myFixture = TestBed.createComponent(HeroesComponent);
    expect(myFixture).toMatchSnapshot();
  });
  it('should render my heroes title', () => {
    const fix = TestBed.createComponent(HeroesComponent);
    const elem = fix.debugElement.componentInstance;
    expect(elem.heroTitle).toEqual('My Heroes');
  });
});
