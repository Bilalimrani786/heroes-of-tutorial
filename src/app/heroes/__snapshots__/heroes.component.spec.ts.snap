// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesComponent should create 1`] = `
<app-heroes
  heroService={[Function HeroService]}
  heroTitle={[Function String]}
>
  <h2>
    
  </h2><ul
    class="heroes"
  >
    
  </ul>
</app-heroes>
`;
